<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Data Integration in Bioinformatics and Biomedicine</title>
    <meta charset="utf-8" />
    <meta name="author" content="Alex Sanchez-Pla, Francesc Carmona, Ferran Reverter, Esteban Vegas, Pol Castellano" />
    <meta name="date" content="2021-07-08" />
    <script src="libs/header-attrs/header-attrs.js"></script>
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="mycss.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Data Integration in Bioinformatics and Biomedicine
### <strong><u>Alex Sanchez-Pla</u></strong>, Francesc Carmona, Ferran Reverter, Esteban Vegas, Pol Castellano
### Genetics, Microbiology and Statistics Department (UB) <br> Statistics and Bioinformatics Unit (VHIR)
### 2021-07-08

---





# Outline

.columnwide[
  ### 1) [Data, Data-related Projects and Data Science](#Introduction)
  ### 2) [Integrative Omics Data Analysis](#IODA)
  ### 3) [Open problems we work in](#IODAProjects)
  ### 4) [Biomedical data integration and data sharing](#DataSharing)
  ### 5) [Data Projects we work in](#DataProjects)
  ### 6) [Looking ahead](#Estrategia)
]

---

class: inverse, middle, center

name: Introduction

# Data, Data-related Projects and &lt;br&gt; Data Science &lt;a id="Introduction"&gt;&lt;/a&gt;

---

# The importance of Data 

## Data is essential to solve problems or answer questions

- Simplistically, when we wish to solve a problem we collect data

    - Statistics:     Questions --&gt; [Hypothesis] --&gt; Get data
    - Data Science:   Questions --&gt; Get Data

- Data availability and complexity increases all the time

    - More (Big) data --&gt; Better solutions? 
    
- 21st century holy grial:

    - __Data + AI --&gt; Personalized medicine__
    
&lt;!-- --- --&gt;

&lt;!-- # How did we get here? --&gt;

&lt;!-- - First we took traditional approach --&gt;

&lt;!-- - Then we made the leap into microarrays --&gt;

&lt;!-- - Next we moved into-multi-omics --&gt;

&lt;!-- - Now we look forward to combining it with clinical data --&gt;

---

# From XXth century studies

## Traditional Statistical Analysis

.center[ 
&lt;img align = "center" src="images/diseaseStudyClassic.png" width = "90%"&gt; 
 ] 

---

# To beginning of XXth 

## Microarrays: One dataset, _many_ genes

### Efron called the XXIst century "_The century of microarrays_"

.center[ 
&lt;img align = "center" src="images/diseaseStudySingleOmics.png" width = "80%"&gt;
 ] 

---
# Why integration?

## The whole is more than the sum of parts

.center[ 
&lt;img align = "center" src="images/elephant.png" width = "90%"&gt; 
 ]

---

# Why integration?

## We May lose important information

.center[ 
&lt;img align = "center" src="images/mayMissSomething.jpg" width = "90%"&gt; 
 ]

---
# Why integration?

## Don't forget the Omics Cascade

.center[ 
&lt;img align = "center" src="images/omicsCascade.png" width = "90%"&gt; 
 ]

---
# From single to multi-omics

## Integromics entered the game (_We want it all_)

.center[ 
&lt;img align = "center" src="images/diseaseStudyMultiOmics.png" width = "90%"&gt; 
 ]

---

# Can we get more data?

## From MultiOmics to Personalized Medicine


.center[ 
&lt;img align = "center" src="images/fromMultiOmics2PM.jpg" width = "90%"&gt; 
]

Google "multi-omics for personalized medicine" images to see more ...

---

# Data-Centric Projects have become key

- The need to have multimodal data, including omics data, images, and clinical data &lt;br&gt;
has driven a multitude of __Research Projects__ &lt;br&gt;

- Their first or ultimate objective is to _facilitate the path to personalized medicine_ through:
.pull-left[

  - data integration 
  - data sharing, 
  - data exchange (OMOP, HL7)
  - data FAIRification,
  - cloud computing and 
  - (federated) data analysis,&lt;br&gt;
among others.
  ]
  
.pull-right[
&lt;img align = "center" src="images/DataProjects2.png" width = "100%"&gt; 
]

&lt;p&gt;  
  
- This represents an _unprecedented opportunity_.
- But requires the right people and knowledge.

---


class: inverse, middle, center

name: IODA

# Integrative Omics Data Analysis

---

# The goal(s) of Integrative Omics

- The idea that efficient integration of data from different OMICS can facilitate the discovery of true causes and states of disease has pervaded the biomedical community.

- The general gol of integrative analysis is the _deciphering of complex biological relationships (CBP)_ empowered by the _combined use of distinct pieces of information_ that represent a, probably partial, view of the _different levels at which these processes happen_.

- More specifically integrative omics data anlysis is applied for
    - Disease subtyping,
    - Disease insights,
    - Biomarker development,
    - Combination with non-omics

---

# Methods of integrative analysis

.center[ 
&lt;img align = "center" src="images/IODAMethods2020.png" width = "70%"&gt; 
]

Subramanian et alt., 2020. _Multi-omics Data Integration, Interpretation, and Its Application_


---

# Many data repositories

.center[ 
&lt;img align = "center" src="images/IODADataRepositories2020.png" width = "90%"&gt; 

Subramanian et alt., 2020. _Multi-omics Data Integration, Interpretation, and Its Application_
]

---

# Many data visualization portals

.center[ 
&lt;img align = "center" src="images/IODAVisualizationPortals.png" width = "80%"&gt; 

Subramanian et alt., 2020. _Multi-omics Data Integration, Interpretation, and Its Application_
]

---

# So what?

- We willl restrict to arbitrarily chosen situations:

    - Multivariate statistical methods, classic and extensions 

- for which we will sketch,

    - General ideas
    - Use case
    
- and provide some examples of use.

    - See workshop

---

# General framework: Matrix factorization

- Matrix factorizations have become very popular in fields such as machine learning, recommender systems or deep learniong.

.center[ 
&lt;img align = "center" src="images/MatrixFactorization-Strang.png" width = "80%"&gt; 
]

_"Many key ideas of linear algebra, when you look at them closely, are really factorizations of a matrix, where the original matrix becomes the product of 2 or 3 special matrices."_ &lt;br&gt;
Gilbert Strang.

---
# Omics technolgies yield data matrices

.center[ 
&lt;img align = "center" src="images/omicsMatrix.png" width = "100%"&gt; 
]

---

# Matrices can be factored

-  Factorization helps discover the "true" biological dimension &lt;br&gt;
-  by revealing hidden complex biological processes (CBPs)

.center[ 
&lt;img align = "center" src="images/omicsMatrixFactorization.png" width = "80%"&gt; 
]

---
# The SVD

## The mother of all factorizations

.center[ 
&lt;img align = "center" src="images/SVDcartoon.png" width = "90%"&gt; 
]

---
# The SVD

## SVD can be re-written as latent factors

.center[ 
&lt;img align = "center" src="images/SVDasWeights.png" width = "90%"&gt; 
]

---

# Factorization helps interpretation

.center[ 
&lt;img align = "center" src="images/omicsMatrixFactorization2.png" width = "100%"&gt; 
]

---

# Factorization helps interpretation

.center[ 
&lt;img align = "center" src="images/omicsMatrixFactorization3.png" width = "100%"&gt; 
]

---

# Distinct factorizations, distinct views

.center[ 
&lt;img align = "center" src="images/DistinctPatterns.png" width = "100%"&gt; 
]

---

# When a single omics is not enough

- Ideally matrix factorizations can provide dimension-reduced
data visualizations that help detect distinct patterns in
features or samples.

- Sometimes the information in the data does not allow for this
separation, but _extending the factorization to different omics that can be related differently with the latent factors can do the job_.

- As could be expected there exist many ways to do multiple factorization.

  - Multiple Factor Analysis, 
  - Regularized Generalized Canonical Correlation Analysis, 
  - Multiple Coinertia Analysis, ...

---

# Multiple factor Analysis

## Straightforward generalization of PCA

.center[ 
&lt;img align = "center" src="images/multipleFactorAnalysis.png" width = "100%"&gt; 
]
---

# A Colon Cancer Example

- A set of 121 tumors from the TCGA (Weinstein, Collisson, Mills, et al. 2013) colorectal cancer cohort is analyzed. 

- The tumors have been profiled for 
    - gene expression using RNA-seq,
    - mutations using Exome-seq, and 
    - copy number variations using genotyping arrays.
    
- Although two tumors arise in the colon, they may have distinct molecular profiles, which is important for treatment decisions.
- The subset of tumors used in this chapter belong to two distinct molecular subtypes CMS1, CMS3

---

# A Colon Cancer Example

## Single omics don't separate well

.center[ 
&lt;img align = "center" src="images/heatmapsMultiomicsColonCancerStudy1.png" width = "90%"&gt; 
]

---

# A Colon Cancer Example

## Joint omics factorization provides much better separation

.center[ 
&lt;img align = "center" src="images/multipleFactorAnalysisInColonCancer.png" width = "100%"&gt; 
]

---

# Summary part I

- Biological Processes are complex and they are "probably" better revealed using distinct sources of information.

- While there are many approache for the integrative analysis there is no universal "IODA" method.

  - Many families of many types of methods available: Need to be related, classified, filtered, benchmarked.
  - In many situations biology must guide the analysis
  - All data are not equally informative and it is often the case that some omics dominate others. : "Gene expression and _what else_?"

---

# The workshop

- Details and examples of single and multi-omics analyses are provided in the workshop.

- Access [https://github.com/alexsanchezpla/202107-GRBio-Integration-Workshop](https://github.com/alexsanchezpla/202107-GRBio-Integration-Workshop) for the workshop materials.

  - You can clone or download the repository (see previous GRBio workshops)
  - Run the `InstallPackages.R` script in your computer to be sure all packages can be installed

- __Warning__ the materials will be updated. Be sure to refressh the repository tomorrow at 9:30



---


class: inverse, middle, center

name: IODAProjects

# Integration problems we work with

---
# FOBI (Food-Biomarker Ontology)
### Pol Castellano, Alex Sánchez

 .font130[
  - Heterogeneus nutritional data (&lt;u&gt;semantic problem&lt;/u&gt;) -&gt; **FoodOn**
  ]    

  .font130[
  - Difficult association of nutritional data with other types of data (&lt;u&gt;semantic and quantitative problem&lt;/u&gt;)  
  ]

  .font130[
  - **Unclear relationships between foods and metabolites**     
  ]
  
.center[&lt;img src="pix/foods&amp;metabolites.png" width = "80%"&gt;]


---
# FOBI (Food-Biomarker Ontology)
### Pol Castellano, Alex Sánchez

  .font130[
  - Create an ontology that clearly defines the many complex relationships between **diet derived metabolites** and **foods** in a consistent and homogeneous way
  ]

  .font130[
  - Reuse previous existing terms to maintain a consistent and standardized nomenclature (OBOFoundry)
    - FoodOn 
    - ChEBI
  ]

.center2[&lt;img src="http://www.obofoundry.org/images/foundrylogo.png" width="100"&gt;]    

  .font130[
  - Propose a consistent starting point for nutrimetabolomic studies 
    - Design
    - Validation
  ]

---

# FOBI (Food-Biomarker Ontology)
### Pol Castellano, Alex Sánchez

&lt;a href="https://github.com/pcastellanoescuder/FoodBiomarkerOntology"&gt;&lt;i class="fab fa-github"&gt;&lt;/i&gt;&amp;nbsp; https://github.com/pcastellanoescuder/FoodBiomarkerOntology&lt;/a&gt;&lt;br&gt;

.pull-left[

## Metrics 

  - 2 sub-ontologies
  - 1197 terms
  - 4 different properties
  - 13 food top-level classes
  - 11 biomarker top-level classes
  - More than 4700 relationships
  - Part of **OBOFoundry project** [http://purl.obolibrary.org/obo/fobi.owl](http://purl.obolibrary.org/obo/fobi.owl)
  - FOBI IDs are indexed into the [HMDB](https://hmdb.ca) (Human Metabolome Database) and [FooDB](https://foodb.ca) (Food Database)
]

.pull-right[&lt;img src="pix/200630_FOBI.png" width="700"&gt;]      

---

# Insights in Multi-Group Data Analysis
### Carolina Millapan, Ferran R., Esteban V.

&lt;p&gt;
.pull-left[
- Analysis of variables observed in a set of individuals that belong to dierent groups. 
  - Monitoring metabolite levels in blood of patients in different healthy/disease
conditions,
- Measurements of heavy metals in soil samples from dierent ecological environments, 
- Studies of gene expression for individuals with dierent experimental conditions
]

.pull-right[

&lt;img align = "center" src="images/multiGroup.png" width = "50%"&gt; 

]

---

# Insights in Multi-Group Data Analysis
### Carolina Millapan, Ferran R., Esteban V.

### Multigroup PCA

- __Multigroup Principal Component Analysis__ 

  - Aim:  _maximization of the interdistances between the points of the different groups_.

- __Multi-Group Dimension reduction__

  - Aim: find an orthogonal set of vectors such that each vector v tends to avoid that the projections of the observations of a given group do not overlap the centroids of the projections of the observations of each of the remaining
groups.

---

# Data Integration with Annotations
### Ferran Briansó, Alex Sánchez

&lt;img align = "center" src="images/integrationWithAnnotations1.png" width = "90%"&gt; 


---

# Data Integration with Annotations
### Ferran Briansó, Alex Sánchez

&lt;img align = "center" src="images/integrationWithAnnotations2.png" width = "90%"&gt; 


---

# Data Integration with Annotations
### Ferran Briansó, Alex Sánchez

&lt;img align = "center" src="images/integrationWithAnnotations3.png" width = "90%"&gt; 

---

# The goProfiles Approach

## Pablo Flores, Jordi Ocaña, Miquel Salicrú i Álex Sánchez

&lt;img align = "center" src="images/SimpleFunctionalProfile4cat.png" width = "70%"&gt; 

---

class: inverse, middle, center

name: DataSharing

# Biomedical data integration &lt;br&gt; and data sharing

---

# Precision or Personalized Medicine

- Two patients with the same disease can respond very differently to the same treatment. 

- Why? Mainly due to the different genomic characteristics of each one. 

- Precision or Personalized Medicine aims at helping physicians to select the treatments that are most likely to help patients based on their genetics.

- PM is currently experiencing significant advances due to the appearance of new diagnostic and computer methods that provide an understanding of the molecular bases of the disease, particularly of the genomics. 

---

# Personalized Medicine at a Scale

## The need for a paradigm change

- Until recently, PM  has been associated with experimental therapies _far from daily clinical practice_ 

- But its central elements arae now much easier to reach:

  - Increasing availability of Genomic Data, fast and cheap to produce
  - Increasing computing power in-house or in the cloud (EOSC)
  - Possibility of accessing clinical data in anonymized possibly federated ways
  
- Having the ability to __integrate__ genomic, clinical and other social, environmental and behavioral factors, will (should) lead to a transformation in the way in which decisions are made with direct effect on clinical practice and public health measures.
  &lt;!-- - improving diagnosis,  --&gt;
  &lt;!-- - improving treatments or  --&gt;
  &lt;!-- - increasing the ability to predict the development of diseases. --&gt;

- The ultimate goal is a safer, more efficient, preventive, and predictive medicine. 

---

# How will this be done (in Spain)

## The IMPaCT program

- In mid-September 2020, the Council of Ministers approved the call for granting grants for the __Precision Medicine Infrastructure associated with Science and Technology__ (__IMPaCT__) of the Strategic Action in Health 2017-2020.

- This decision, which involved the granting of 25.8 million euros to the Carlos III Health Institute (ISCIII)

- The programs included in IMPaCT are aligned with several areas that will be developed in the future national strategy:

  - Predictive medicine; 
  - Genomic medicine and 
  - Data science. 
  
---

# The Predictive Medicine Program

.pull-left[

- Aimed at the design and implementation of _a large population cohort with clinical, epidemiological and biological data, measured at the individual level_, 

- which allows representing the entire population residing in Spain, 

- including the ethnic variability and geographic and environmental diversity.

- __Data sharing__ is an essential aspect: clinical data needs to be accessible for this program to be possible.

&lt;!-- This will make it possible to build predictive models of disease, identify health inequalities, monitor key indicators and evaluate the impact of health policies ”, continued Raquel Yotti. --&gt;

]

.pull-right[
&lt;img align = "center" src="images/PredictiveMedicine.jpg" width = "100%"&gt; 

]
---

# The Genomic Medicine Program

.pull-left[

- It will develop _coordination infrastructures and protocols_ to carry out genomic analyzes and other 'omic' data in an effective, efficient and equitable accessible manner.

- It will take as support large research centers that already have of state-of-the-art sequencing technology and experience in its application to the diagnosis of human diseases. 

- It will optimize and reinforce the available massive sequencing capacities, orienting them to the needs of genetic diagnosis -exomes, complete genomes, etc.- 
]

.pull-right[
&lt;img align = "center" src="images/GenomicMedicine.jpg" width = "100%"&gt; 

]
---

# The Data Science Program

- _Develop a system for the collection, integration and analysis of clinical and molecular data__
aimed at improving the health of each individual patient, and 

- Which allows the secondary use of existing information in the SNS to the benefit of society with the objectives of public health, health planning and research. 

- Its objective is to __optimize the management of the information generated__, in order to apply it in the most effective way for the population and the SNS. 

- It will provide 

  - _bioinformatics tools_ for the management of genomic data and 
  - _medical informatics_ solutions for the _management_ and _integration_ of clinical data, 
  
facilitating the interoperability of the clinical information systems of the different autonomous health systems.

---

# IMPaCT Global View


.center[ 
&lt;img align = "center" src="images/IMPACTStructure.png" width = "100%"&gt; 
]

---

# New Skills: 

&lt;p&gt;


## data sharing, 
  
## data exchange (OMOP, HL7)
  
## data FAIRification,
  
## cloud computing, 
  
## open science,
  
## reproducibility,
  
## (federated) data analysis,
  
## and, of course AI and ML


---

class: inverse, middle, center

name: DataProjects

# Data Projects we work in @VHIR

---

# Data Science Projects Platform @VHIR
- The increasing needs in Data related projects leads to new structures to manage this.

.center[ 
&lt;img align = "center" src="images/organigramaUEB.png" width = "70%"&gt; 
 ] 

---

# External Data Projects: EOSC-Life

.center[ 
&lt;img align = "center" src="images/EOSC-Life.png" width = "90%"&gt; 
 ] 

---

# External Data Projects: EU-PEARL

.center[ 
&lt;img align = "center" src="images/EU-PEARL.png" width = "90%"&gt; 
 ] 

---

# External Data Projects: VEIS

.center[ 
&lt;img align = "center" src="images/VEIS.png" width = "90%"&gt; 
 ] 

---

# Other External Data Projects

- The European Health Data Space (TEHDAS)

- EHDEN (European HEalth Data and EVidence Network)

- DARWIN (Data Analysis and Real World Interrogation Network)

All these projects

- Are huge (all over Europe).
- Aim at some form of data re-use for health related research.
- Require different types od (trans-national) data sharing.
- With common or similar technical and legal issues to be addressed

---

class: inverse, middle, center

name: Estrategia

# Looking ahead

- The analysis of Omics Data is an example of fruitful development between Statistics and Biological (Health) Sciences.

- As more data is becoming available some therapeutical possibilities approach reality.

- While Statistics will always be relevant to guide this approach other skills become important.

- We must, at least, be aware of these skills and the problems they address because this is really an unprecedent opportunity.


---


# Acknowledgements

.center[ 
&lt;img align = "center" src="images/acknowledgements2021.png" width = "80%"&gt; 
 ] 


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
